Pocket/pocket-ios
# IN-231: Add Tags Screen Text Field Input

## Summary
Add Tags Screen: user can input a tag into a text field and save it

Note: Small UI updates (i.e. color) will be addressed as soon as designs are ready

## References 
IN-231

## Implementation Details
* Created Add Tags Screen in SwiftUI which uses the `AddTagsViewModel ` that handles tags management for a specific saved item
* Added `Tag` in Core Data Model as a NSOrderedSet (many to many relationship with a SavedItem)
* Updated `list.graphql` with `tags` field and added `replaceSavedItemTags` mutation for updating a saved item's tags
* Created two requests `fetchTags` and `fetchTag(byName name: String)` to be used to update tags locally by first checking if tags already exists
* Updated `SavedItem+RemoteMapping` so that when we call `fetchList` the save item updates retrieves the remote tags and updates the tags data locally
* Updated `PocketSource` so that when we can call `addTags`to send a mutation query to graphQL. This function updates the tags locally with the new data first and then sends the request to the server.
* User can see the Add Tags Screen from a Saved Item in Reader view, My List and Archive

## Test Steps
- [ ] Given a user has tapped on the Add Tags option for a Saved Item,
and the user views the text field (which is attached to the keyboard),
then the user should be able to enter and add tags.

- [ ] Given a user has tapped on the Add Tags option for a Saved Item,
and the user views the text field, 
then the user should be able to see existing tags.

- [ ] Given a user has tapped on the Add Tags option for a Saved Item,
and the user inputs a tag into the text field, 
then the user should be able save it.

- [ ] Given a user has tapped on the Add Tags option for a Saved Item,
and the user taps on a tag, 
then the user should be able to delete it. 

## PR Checklist:

## Screenshots
![Simulator Screen Shot - iPhone 12 mini - 2022-08-24 at 11 55 20](https://user-images.githubusercontent.com/6743397/186465271-1b6256e6-a2bd-4366-8134-95dc6b00ce1c.png)
